chain/gen/genesis/miners.go

```diff
@@ -473,19 +474,36 @@ func SetupStorageMiners(ctx context.Context, cs *store.ChainStore, sys vm.Syscal
 					return cid.Undef, xerrors.Errorf("getting current epoch reward: %w", err)
 				}
 
-				tpow, err := currentTotalPower(ctx, genesisVm, minerInfos[i].maddr)
+				qaps, err := currentQualityAdjPowerSmoothed(ctx, nv, genesisVm, minerInfos[i].maddr)
 				if err != nil {
 					return cid.Undef, xerrors.Errorf("getting current total power: %w", err)
 				}
 
-				pcd := miner9.PreCommitDepositForPower(smoothing9.FilterEstimate(rewardSmoothed), smoothing9.FilterEstimate(*tpow.QualityAdjPowerSmoothed), miner9.QAPowerMax(m.SectorSize))
+				pcd := miner15.PreCommitDepositForPower(smoothing15.FilterEstimate(rewardSmoothed), qaps, miner15.QAPowerMax(m.SectorSize))
 
-				pledge := miner9.InitialPledgeForPower(
+				powerActor, err := nst.GetActor(power.Address)
+				if err != nil {
+					return cid.Undef, xerrors.Errorf("loading power actor: %w", err)
+				}
+
+				powerState, err := power.Load(adt.WrapStore(ctx, cst), powerActor)
+				if err != nil {
+					return cid.Undef, xerrors.Errorf("loading power actor state: %w", err)
+				}
+
+				var epochsSinceRampStart int64
+				if powerState.RampStartEpoch() > 0 {
+					epochsSinceRampStart = -powerState.RampStartEpoch()
+				}
+
+				pledge := miner15.InitialPledgeForPower(
 					sectorWeight,
 					baselinePower,
-					smoothing9.FilterEstimate(rewardSmoothed),
-					smoothing9.FilterEstimate(*tpow.QualityAdjPowerSmoothed),
+					smoothing15.FilterEstimate(rewardSmoothed),
+					qaps,
 					big.Zero(),
+					epochsSinceRampStart,
+					powerState.RampDurationEpochs(),
```

Comment:
Can't you just set this to 0,0 and avoid adding all of this complexity?  This would use the same argument you made in PR comments that differences in pledge are small and there is plenty of buffer.

Reply:
0 epochs since start, 0 ramp duration should put us at the end of the ramp

Reply:
ah you know what, if you don't change it to this then its a bug.  The preseal setup actors stub we left for genesis generation which gets called below assumes ramp time == 0. https://github.com/filecoin-project/builtin-actors/pull/1557/files#diff-fbcb2ec1a9d82b18f146c728cafd643df0e7ae47a04d84be7644913fe89236e5R1988